uint8 x_dim
uint8 y_dim
uint16[] data
